.github/workflows/check_changed_files.yml

```diff
@@ -0,0 +1,51 @@
+name: Check changed files
+
+on: 
+    pull_request:
+        branches:
+            - main
+
+jobs:
+
+    check-changed-files:
+
+        runs-on: ubuntu-latest
+
+        env:
+            TEMPLATE_REPO: neurodatascience/nipoppy
+            PATH_CURRENT_REPO: repo_to_check
+            PATH_TEMPLATE_REPO: template_repo
+
+        if: github.repository_owner == 'neurodatascience' && github.event.repository.name != 'nipoppy'
+
+        steps:
+
+          - name: Checkout current repository
+            uses: actions/checkout@v3
+            with:
+                fetch-depth: 0
+                path: ${{ env.PATH_CURRENT_REPO }}
+
+          - name: Checkout template repository
+            uses: actions/checkout@v3
+            with:
+                repository: ${{ env.TEMPLATE_REPO }}
+                ref: main
+                path: ${{ env.PATH_TEMPLATE_REPO }}
+
+          - name: Check changed files
+            run: |
+                CHANGED_FILES=$(cd ${PATH_CURRENT_REPO}; git diff-tree --no-commit-id --name-only -r ${{ github.event.pull_request.head.sha }} ${{ github.event.pull_request.base.sha }})
```

Comment:
Since I don't speak github actions dialect, just want to check if this looks for entire file path or just the filenames? Because we have multiple `READMEs` :)

Reply:
This part is just like a regular shell script, `git diff-tree` gives full paths (relative to the root of the `git` repository) so it shouldn't be a problem. Tested it [here](https://github.com/neurodatascience/testpoppy-ppmi/pull/3) with the READMEs and it works fine :)